### PR TITLE
Trailing whitespace cleanup

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Comment by dalg24** [comment]
ok to test

**Comment by dalg24** [comment]
ok to test

**Comment by dalg24** [comment]
retest this please

**Comment by dalg24** [comment]
ok to test

File: CMakeLists.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ TRIBITS_ADD_ETI_INSTANTIATIONS(TpetraCore
```
Comment (dalg24) [inline CMakeLists.txt]: We need to add a test that ensures no trailing whitespace was introduced in a new commit.

**Comment by dalg24** [comment]
@sslattery: would you take a look at the [libMesh failing test](https://jupyterdocker.ornl.gov:8080/job/DataTransferKit-experimental/5/testReport/junit/projectroot.DataTransferKit.packages.Adapters.Libmesh/test/DataTransferKitLibmeshAdapters_LibmeshVariableField_test_MPI_4/).  Obviously, Andrey's changes have nothing to do with it.

File: CMakeLists.txt (hunk 1 above)
Comment (aprokop) [inline CMakeLists.txt]: I don't know if it is possible on github (webhooks?), but on client side a simple pre-commit hook is all you need.

**Comment by dalg24** [comment]
BTW checkout [this](https://github.com/blog/967-github-secrets) to see how to ignore whitespace in a commit.  This allows to verify that this PR actually only removes whitespace.

File: CMakeLists.txt [hunk 2]
```diff
@@ -74,5 +74,14 @@ IF(Trilinos_ENABLE_DataTransferKitMoabAdapters)
 ENDIF()
 
 ##---------------------------------------------------------------------------##
+## Check that there are no trailing whitespace
+##---------------------------------------------------------------------------##
+
+ADD_TEST(
+    NAME check_no_trailing_whitespace
+    COMMAND sh -c "grep --exclude=*.git* -rl '[[:blank:]]$' ${CMAKE_SOURCE_DIR} test $? -eq 1"
+)
```
Comment (dalg24) [inline CMakeLists.txt]: Here is one way of doing that.  It will fail right now because you only edited the *.{cpp,cpp} files in your commit.

Comment (aprokop) [inline CMakeLists.txt]: Would this not include the whole Trilinos? Even if you run it in DTK source, would that not include files like pdf if they are there? I think it should only be enforced for some file types, and not for all files.

Comment (dalg24) [inline CMakeLists.txt]: Oh you are right, we need to restrict to DTK source.  We can adjust and exclude any file pattern we like.

Comment (dalg24) [inline CMakeLists.txt]: `${CMAKE_CURRENT_SOURCE_DIR}` or `${${PACKAGE_NAME}_SOURCE_DIR}` would do.

File: CMakeLists.txt [hunk 3]
```diff
@@ -78,8 +78,17 @@ ENDIF()
 ##---------------------------------------------------------------------------##
 
 ADD_TEST(
-    NAME check_no_trailing_whitespace
-    COMMAND sh -c "grep --exclude=*.git* -rl '[[:blank:]]$' ${CMAKE_SOURCE_DIR} test $? -eq 1"
+    NAME check_no_trailing_whitespace_nor_tab
+    COMMAND sh -c "grep \
+         --exclude-dir={.git,doc} \
```
Comment (dalg24) [inline CMakeLists.txt]: I think that's where the error is.  I tested with `.` in place of `${${PACKAGE_NAME}_SOURCE_DIR}`.

**Comment by dalg24** [comment]
The changes were applied to a couple `Makefile`s in `doc/`.  We probably don't want that.

**Comment by dalg24** [comment]
@aprokop: I wasn't thinking and did a push force.  Sorry about that.

**Comment by aprokop** [comment]
@dalg24 you did not! I did the force push later.
Why skip Makefiles? 

Do you know why Travis fails?

**Comment by dalg24** [comment]
> Why skip Makefiles?
Because they are part of the syntax as far as I remember.

> Do you know why Travis fails?
Quantum oscillations...  No I am not sure.

**Comment by aprokop** [comment]
@dalg24 Oh, you are right. You wrote Makefile, I read CMakeLists.txt

**Comment by aprokop** [comment]
This PR closes #76 once merged.

We may wait for formatting PR addressing #77 before merging to get them close to each other.

**Comment by dalg24** [comment]
PR #91 has been merged.  @aprokop would you rebase this on the master?

**Comment by dalg24** [comment]
closes #78
